Tests/UnitTests/Scripts/engine-compat-tests.ts

```diff
@@ -0,0 +1,413 @@
+                    engineName = "V8";
+                }
+            } catch (e) {
+                // Try alternate V8 detection
+                isV8 = typeof (globalThisPolyfill as any).v8 !== 'undefined' ||
+                       typeof (globalThisPolyfill as any).d8 !== 'undefined';
+                if (isV8) {
+                    engineDetected = true;
+                    engineName = "V8";
+                }
+            }
+
+            // JavaScriptCore detection
+            let isJSC = false;
+            if (!isV8) {
+                try {
+                    const funcStr = Function.prototype.toString.call(Math.min);
+                    // JSC format includes newlines in native function representation
+                    isJSC = funcStr.includes("[native code]") && funcStr.includes("\n");
+                    if (isJSC) {
+                        engineDetected = true;
+                        engineName = "JavaScriptCore";
+                    }
+                } catch (e) {
+                    // Fallback JSC detection
+                    if (hostPlatform === "iOS" || hostPlatform === "Darwin") {
+                        isJSC = true;
+                        engineDetected = true;
+                        engineName = "JavaScriptCore";
+                    }
+                }
+            }
+
+            // Chakra detection for Windows
+            const isChakra = !isV8 && !isJSC && hostPlatform === "Win32";
+            if (isChakra) {
+                engineDetected = true;
+                engineName = "Chakra";
+            }
+
+            // If no engine detected through specific checks, use a fallback
+            if (!engineDetected) {
+                // On Android, if not JSC, assume V8 (most common)
+                if (hostPlatform === "Android") {
+                    isV8 = true;
+                    engineDetected = true;
+                    engineName = "V8 (assumed)";
+                }
+            }
+
+            console.log(`Engine: ${engineName}`);
+            console.log(`Platform: ${hostPlatform}`);
+
+            // At least one engine should be detected
+            expect(engineDetected).to.be.true;
+        });
+
+        it("should report engine version if available", function () {
+            // V8 version check
+            if (typeof (globalThisPolyfill as any).v8 !== 'undefined') {
+                try {
+                    const version = (globalThisPolyfill as any).v8.getVersion?.();
+                    if (version) {
+                        console.log(`V8 Version: ${version}`);
+                        expect(version).to.be.a('string');
+                    }
+                } catch (e) {
+                    // Some V8 builds might not expose version
+                }
+            }
+            // If no version info, just pass the test
+            expect(true).to.be.true;
+        });
+    });
+
+    describe("N-API Compatibility", function () {
```

Comment:
How are these tests related to N-API? I don't see any code here that is related to N-API. Can you explain?